### PR TITLE
Refactored Site Classification methods

### PR DESCRIPTION
**Pull request by erwinvanhunen**

### CODE REVIEW

File: Core/OfficeDevPnP.Core/Extensions/TenantExtensions.cs [hunk 1]
```diff
@@ -916,81 +858,19 @@ public static IEnumerable<String> GetSiteClassificationList(this Tenant tenant,
         /// <param name="classificationList">The list of classification values</param>
         /// <param name="defaultClassification">The default classification</param>
         /// <param name="usageGuidelinesUrl">The URL of a guidance page</param>
-        public static void UpdateSiteClassification(this Tenant tenant, String accessToken, IEnumerable<String> classificationList, String defaultClassification = null, String usageGuidelinesUrl = null)
+        public static void UpdateSiteClassificationSetings(this Tenant tenant, String accessToken, IEnumerable<String> classificationList, String defaultClassification = null, String usageGuidelinesUrl = null)
```
Comment (PaoloPia) [inline Core/OfficeDevPnP.Core/Extensions/TenantExtensions.cs]: Plese fix the type UpdateSiteClassificationSettings (double t)

File: Core/OfficeDevPnP.Core/Framework/Graph/SiteClassificationUtility.cs [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+using OfficeDevPnP.Core.Framework.Graph.Model;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OfficeDevPnP.Core.Framework.Graph
+{
+    public static class SiteClassificationUtility
```
Comment (PaoloPia) [inline Core/OfficeDevPnP.Core/Framework/Graph/SiteClassificationUtility.cs]: I would rename this one to SiteClassificationsUtility, if we made plural names in the extension methods, I would keep this one plural, too.

File: Core/OfficeDevPnP.Core/Framework/Graph/SiteClassificationUtility.cs [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+                throw new ApplicationException("Missing DirectorySettingTemplate for \"Group.Unified\"");
+            }
+        }
+
+        /// <summary>
+        /// Enables Site Classification for the target tenant 
+        /// </summary>
+        /// <param name="accessToken">The OAuth accessToken for Microsoft Graph with Azure AD</param>
+        /// <returns>The list of Site Classification values</returns>
+        public static SiteClassificationSettings GetSiteClassificationSettings(String accessToken)
```
Comment (PaoloPia) [inline Core/OfficeDevPnP.Core/Framework/Graph/SiteClassificationUtility.cs]: Same here? Classifications or Classification?

File: Core/OfficeDevPnP.Core/Framework/Graph/Model/SiteClassification.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OfficeDevPnP.Core.Framework.Graph.Model
+{
+    /// <summary>
+    /// Represents settings with regards to Site Classification
+    /// </summary>
+    public class SiteClassificationSettings
```
Comment (PaoloPia) [inline Core/OfficeDevPnP.Core/Framework/Graph/Model/SiteClassification.cs]: Classification or Classifications?

File: Core/OfficeDevPnP.Core/Framework/Graph/SiteClassificationUtility.cs [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+                throw new ApplicationException("Missing DirectorySettingTemplate for \"Group.Unified\"");
+            }
+        }
+
+        /// <summary>
+        /// Updates Site Classification settings for the target tenant
+        /// </summary>
+        /// <param name="accessToken">The OAuth accessToken for Microsoft Graph with Azure AD</param>
+        /// <param name="siteClassification">The site classification settings to apply./param>
+        public static void UpdateSiteClassificationSettings(string accessToken, SiteClassificationSettings siteClassification)
```
Comment (PaoloPia) [inline Core/OfficeDevPnP.Core/Framework/Graph/SiteClassificationUtility.cs]: Classification or Classifications?

File: Core/OfficeDevPnP.Core/Framework/Graph/SiteClassificationUtility.cs [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+        }
+
+        /// <summary>
+        /// Updates Site Classification settings for the target tenant
+        /// </summary>
+        /// <param name="accessToken">The OAuth accessToken for Microsoft Graph with Azure AD</param>
+        /// <param name="classificationList">The list of classification values</param>
+        /// <param name="defaultClassification">The default classification</param>
+        /// <param name="usageGuidelinesUrl">The URL of a guidance page</param>
+        public static void UpdateSiteClassificationSettings(String accessToken, IEnumerable<String> classificationList = null, String defaultClassification = "", String usageGuidelinesUrl = "")
```
Comment (PaoloPia) [inline Core/OfficeDevPnP.Core/Framework/Graph/SiteClassificationUtility.cs]: Classification or Classifications?

**Comment by VesaJuvonen** [comment]
thx Erwin for your submission, we'll have a look on this asap.

**Pull request merged**